Formula/chain-bench.rb

```diff
@@ -0,0 +1,18 @@
+class ChainBench < Formula
+  desc "Software supply chain auditing tool based on CIS benchmark"
+  homepage "https://github.com/aquasecurity/chain-bench"
+  url "https://github.com/aquasecurity/chain-bench/archive/v0.1.3.tar.gz"
+  sha256 "46b63528ddd0e09b65539a69aefbf4229343e636854d0dd5a241870717843423"
+  license "Apache-2.0"
+  head "https://github.com/aquasecurity/chain-bench.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-s -w -X=main.version=#{version}"), "./cmd/chain-bench"
+  end
+
+  test do
+    system "#{bin}/chain-bench", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hey @SMillerDev, I have conducted an assert failure test since without credentials the run of the CLI will fail in any case.

And maintained the version test also.